subsys/bluetooth/mesh/access.c

```diff
@@ -931,12 +947,10 @@ void bt_mesh_comp_provision(uint16_t addr)
 	LOG_DBG("addr 0x%04x elem_count %zu", addr, dev_comp->elem_count);
 
 	for (i = 0; i < dev_comp->elem_count; i++) {
-		struct bt_mesh_elem *elem = &dev_comp->elem[i];
-
-		elem->addr = addr++;
```

Comment:
I guess you missed the element address initialization. It is used actively in self-provisioning and restoring device data from cdb.

Reply:
I don't quite understand what you mean

Reply:
This PR broke the addressing of elements. Now the function initializes only the primary element. Please fix it.

Reply:
What does this function do from your point of view?

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/f66b14974268b9692980317dfb4663a03aee949b/subsys/bluetooth/mesh/access.c#LL943C6-L943C30

Reply:
> 5 Provisioning
> Provisioning is a process of adding an unprovisioned device to a mesh network, managed by a
> Provisioner. A Provisioner provides the unprovisioned device with provisioning data that allows it to
> become a mesh node. The provisioning data includes a network key, the current IV Index, **and the unicast
> address for each element.**

Please take a look at specification.
Yep, configuring global variable is mandatory but not sufficient. You removed assigning of the unicast addresses for each element.

Reply:
**`the unicast address for each element`** In my opinion, obtaining through API has the same behavior, which does not violate the specification.

Reply:
**2.3.4 Elements**

The primary element is addressed using the first unicast address assigned to the node during
provisioning. Each additional secondary element is addressed using the subsequent addresses. These
unicast element addresses allow nodes to identify which element within a node is transmitting or receiving
a message.

**5.4.2.5 Distribution of provisioning data**

Upon receiving the Provisioning Data PDU from the Provisioner, the device shall decrypt and authenticate
the provisioning data. Upon successful authentication of the provisioning data, the device shall set the
network key (identified by the Key Index), set the IV index, set the IV Update procedure state based on
the IV Update Flag, set the Key Refresh Phase based on the Key Refresh Flag, and assign unicast
addresses to all device elements starting from primary element using consecutive range of addresses
starting from provided unicast address value.

From the perspective of code implementation, since it has been established as a consecutive change, why do we have to keep every record instead of using operations?

Reply:
Yep, correct. Exactly 2.3.4 spec chapter is broken now by this PR if provision device over API (not over provisioner).

Reply:
@alxelax could you elaborate on what is actually broken? For example, use case or code sequence which is broken. From what I see, any code that used `address` field of `bt_mesh_elem` will now use `bt_mesh_elem_addr_get` function which will always give actual address.

Reply:
I still don't get you. Why do you think that only primary element gets address? When a user needs to know elements address, it calls `bt_mesh_elem_addr_get()`. `bt_mesh_elem` structure doesn't have the address field anymore.

Reply:
Ok, I see. Why is this cycle required at all?

Reply:
Just to print debug information like it was doing before.

Reply:
before it was element address initialization.
Now it is useless. Address is calculated dynamically. Composition data is created by application and provided to mesh stack. 
Why should mesh print data that application already knows?

Reply:
@LingaoM ?

Reply:
sorry, I'm not understand.

Reply:
Do you think we need to keep these debug messages or can we remove them?

Reply:
Understand.